pkg/state/process_unix.go

```diff
@@ -0,0 +1,38 @@
+//go:build aix || darwin || dragonfly || freebsd || linux || netbsd || openbsd || solaris || zos
+// +build aix darwin dragonfly freebsd linux netbsd openbsd solaris zos
+
+package state
+
+import (
+	"fmt"
+	"os"
+	"syscall"
+)
+
+func pidExists(pid int) (bool, error) {
+	if pid <= 0 {
+		return false, fmt.Errorf("invalid pid %v", pid)
+	}
+	proc, err := os.FindProcess(pid)
```

Comment:
Wondering why we have a different implementation on Unix? `os.FindProcess` does not already check and return the right info about the process?

Reply:
Ah, got it. From the [doc](https://pkg.go.dev/os#FindProcess), it looks like this is expected:

> On Unix systems, FindProcess always succeeds and returns a Process for the given pid, regardless of whether the process exists.

Reply:
Yes, this is the reason